server/server.js

```diff
@@ -143,6 +164,183 @@ app.get("/api/signout", (c) => {
   return c.json({ message: "Successfully signed out." });
 });
 
+app.get("/api/records", async (c) => {
+  const param = await c.req.json();
+
+  if (!param.start || !param.end) {
+    throw new HTTPException(400, { message: "parameters \"start\" and \"end\" are required." });
+  }
+
+  // 型バリデーション
+  if (typeof param.start !== "string" || typeof param.end !== "string") {
+    throw new HTTPException(400, { message: "parameters \"start\" and \"end\" must be string." });
+  }
+
+  const token = getCookie(c, COOKIE_NAME);
+
+  if (!token) {
+    throw new HTTPException(401, { message: "Unauthorized" });
+  }
+
+  // UTCに変換してからデータベースから取得
+  const utcStart = new Date(param.start).toISOString();
+  const utcEnd = new Date(param.end).toISOString();
+
+  const records = db.prepare(queries.DrinkingRecords.findByUserIdAndDateRange).all(userId, utcStart, utcEnd);
+
+  return c.json(records);
+});
+
+// 飲酒量記録のエンドポイント
+app.post("/api/records", async (c) => {
+  const param = await c.req.json();
+
+  // パラメータが存在するか確認
+  if (!param.date || !param.amount || !param.condition) {
+    throw new HTTPException(400, { message: "date, amount, and condition are required." });
+  }
+
+  // 型バリデーション
+  if (typeof param.date !== "string") {
+    throw new HTTPException(400, { message: "date must be a string." });
+  }
+
+  if (typeof param.amount !== "number") {
+    throw new HTTPException(400, { message: "amount must be a number." });
+  }
+
+  if (typeof param.condition !== "number") {
+    throw new HTTPException(400, { message: "condition must be a number." });
+  }
+
+  // JWTからユーザーIDを取得
+  const token = getCookie(c, COOKIE_NAME);
+  if (!token) {
+    throw new HTTPException(401, { message: "Unauthorized" });
+  }
+
+  const userId = await getUserIdFromJwt(token, JWT_SECRET);
+
+  // JSTで日付のバリデーション
+  const selectedDate = new Date(param.date);
+
+  const minDateJst = new Date("2000-01-01T00:00:00Z");
+
+  const nowDateJst = new Date();
+  nowDateJst.setHours(nowDateJst.getHours() + 9); // JSTとして解釈するために時差を足す
+
+  if (selectedDate < minDateJst || selectedDate > nowDateJst) {
+    throw new HTTPException(400, { message: "date must be between 2000-01-01 (JST) and today." });
+  }
+
```

Comment:
```suggestion
//飲酒量のバリデーション
if (param.amount < 0) {
  throw new HTTPException(400, { message: "amount must  be greater than or equal to 0." });
}
```
動作確認しているときに思ったんですが、クライアント側で`amount`が0以上かどうかバリデーションしているのでもしかしたらサーバー側でもやっていいのかなと思います！